libyul/backends/evm/EVMMetrics.cpp

```diff
@@ -87,14 +87,16 @@ void GasMeterVisitor::operator()(FunctionCall const& _funCall)
 
 void GasMeterVisitor::operator()(Literal const& _lit)
 {
-	m_runGas += evmasm::GasMeter::runGas(evmasm::Instruction::PUSH1, m_dialect.evmVersion());
-	m_dataGas +=
-		singleByteDataGas() +
-		evmasm::GasMeter::dataGas(
-			toCompactBigEndian(valueOfLiteral(_lit), 1),
-			m_isCreation,
-			m_dialect.evmVersion()
-		);
+	m_runGas += evmasm::GasMeter::pushGas(valueOfLiteral(_lit), m_dialect.evmVersion());
+
+	if (!m_dialect.evmVersion().hasPush0() || valueOfLiteral(_lit) != u256(0))
+		m_dataGas +=
+			singleByteDataGas() +
+			evmasm::GasMeter::dataGas(
+				toCompactBigEndian(valueOfLiteral(_lit), 1),
+				m_isCreation,
+				m_dialect.evmVersion()
+			);
```

Comment:
In the `else` case we still need one byte for the ``PUSH0`` opcode itself, don't we?
Unless I'm missing something, this would still need to change.

Reply:
Yes, I think that I was confused about what these costs were. 
Fixed it.